### ISSUE TITLE
Writing of unit tests

### ISSUE THREAD

**Issue by christianlupus** [nextcloud/cookbook#298]
**Is your feature request related to a problem? Please describe.**
Currently there is a very basic framework to do unit testing in place. However, most parts of the code base are untested. This makes any PR with more than a few lines of changes code very hard to check.

**Describe the solution you'd like**
Extend the current structure of tests to the whole code base and run these tests (including code coverage checks) on each commit.

**Describe alternatives you've considered**
The only alternative is to decide not to do any tests and rely on manual testing and hoping for the best.

**Comment by TobiasMie** [comment]
Hi,
I would like to help writing tests. 
Is there something I should pay special attention to? Or a part that needs to be tested first?

**Comment by christianlupus** [comment]
I am currently planning on reworking on the CI development see #296 , #27, #233. As no CI is running currently, let me do this DevOp first.

My ideal solution would be to have a docker image ready to be used for local unit testing.

Apart from that I do not see much that needs to be considered now.

**Comment by christianlupus** [comment]
@TobiasMie I opened PR #337. This might be of interest for you if you are still willing to write unit tests. Your help would really be appreciated, especially as we have in fact no tests. I needed to remove all during the PR as the classes they should test were changed heavily since the tests were working.
I hope that the PR can be merged soon allowing us to write real unit tests. Until then you might want to start from the [dev/action branch](https://github.com/christianlupus/nextcloud-cookbook/tree/dev/actions) to have at least a running structure. Sorry for the missing documentation, it is still on my agenda.

**Comment by christianlupus** [comment]
@TobiasMie as the PR has been merged in the meantime, I wanted to ask what your state is. Are you still interested in unit-testing? Have you even started working?

**Comment by TobiasMie** [comment]
Yes I'm still interested. I've run into some issues setting up the environment but the docker solution works great. I will work on some tests in the next days but I cannot promise much as university has started.

**Comment by christianlupus** [comment]
That is great to hear.

I am on my way to do some significant update to the codebase. I will try get a grip today. The question is: Do you want to work on the old codebase (possibly causing the tests to be rewirtten), directly on the new codebase, or better wait for the university to settle down while the refactory takes place (which could be quite some time)?
I wanted to sync with you before I start tweaking around in the codebase. For quick discussions: Are you in the matrix channel, @TobiasMie?

**Comment by TobiasMie** [comment]
I think it makes more sense to work directly on the new code instead of rewriting tests. If you tell me witch parts of the new code are more or less done I can start with them. 
I've joined the matrix channel just now.
